trillian/ctfe/handlers.go

```diff
@@ -230,32 +230,40 @@ type logInfo struct {
 	validationOpts CertValidationOpts
 	// rpcClient is the client used to communicate with the trillian backend
 	rpcClient trillian.TrillianLogClient
-	// signer signs objects
+	// signer signs objects (e.g. STHs, SCTs) for regular logs
 	signer crypto.Signer
-
-	// Cache the last signature generated for an STH, to reduce re-signing and
-	// slightly reduce the chances of being able to fingerprint get-sth users by
-	// their STH signature value.
-	sthSigCache SignatureCache
+	// sthGetter provides STHs for the log
+	sthGetter STHGetter
 }
 
 // newLogInfo creates a new instance of logInfo.
-func newLogInfo(logID int64, prefix string, validationOpts CertValidationOpts, rpcClient trillian.TrillianLogClient, signer crypto.Signer, instanceOpts InstanceOptions, timeSource util.TimeSource) *logInfo {
-	ctx := &logInfo{
+func newLogInfo(
+	logID int64, prefix string, isMirror bool, validationOpts CertValidationOpts,
```

Comment:
Too many arguments, but it was there before so we can fix it later.

Reply:
Added a TODO.